arch/avr/src/avr/up_schedulesigaction.c

```diff
@@ -152,13 +153,15 @@ void up_schedule_sigaction(struct tcb_s *tcb, sig_deliver_t sigdeliver)
               /* Then set up to vector to the trampoline with interrupts
                * disabled
                */
+
+              reg_ptr = (uintptr_t)up_sigdeliver;
```

Comment:
We can directly init reg_ptr at line 94, then both line 157 and 201 can be removed.